### PR TITLE
fix(create-gatsby): Respect telemetry disable

### PR DESCRIPTION
**Pull request by tyhopp**
## Description

Ensure `create-gatsby` respects user-provided environment variable to disable telemetry.

## Related Issues

- Fixes https://github.com/gatsbyjs/gatsby/issues/34486
- [sc-44464]

### CODE REVIEW

File: integration-tests/ssr/__tests__/ssr.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ describe(`SSR`, () => {
     expect(String(childProcess.stdout)).toContain(
       `testing these paths for differences between dev & prod outputs`
     )
-  }, 30000)
+  }, 60000)
```
Comment (LekoArts) [inline integration-tests/ssr/__tests__/ssr.js:36]: Let's leave the changes to SSR test out of this and put it into a separate PR :)

File: packages/create-gatsby/src/__tests__/tracking.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    })
+
+    const enabled = isTrackingEnabled()
+    expect(enabled).toBeFalse()
+
+    const cachedEnabled = isTrackingEnabled()
+    expect(cachedEnabled).toBeFalse()
+  })
+
+  describe(`isTrackingEnabled / process.env.GATSBY_TELEMETRY_DISABLED`, () => {
```
Comment (LekoArts) [inline packages/create-gatsby/src/__tests__/tracking.ts:44]:
```suggestion
  describe(`process.env.GATSBY_TELEMETRY_DISABLED`, () => {
```

It'll already be nested in Jest and have `isTrackingEnabled` as "parent"

File: packages/create-gatsby/src/__tests__/tracking.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    beforeAll(() => {
+      process.env.GATSBY_TELEMETRY_DISABLED = `true`
+    })
+
+    it(`respects the setting of the environment variable`, async () => {
```
Comment (LekoArts) [inline packages/create-gatsby/src/__tests__/tracking.ts:49]: We can drop the `async` on each test

File: packages/create-gatsby/src/tracking.ts [hunk 4]
```diff
@@ -28,7 +33,34 @@ export interface ITrackCliArgs {
...
+  if (enabled === undefined || enabled === null) {
+    enabled = true
+    store.set(`telemetry.enabled`, enabled)
+  }
+  trackingEnabled = enabled
+
+  if (trackingDisabledFromEnvVar) {
+    enabled = false
+    trackingEnabled = enabled
+  }
```
Comment (LekoArts) [inline packages/create-gatsby/src/tracking.ts:54]:
```suggestion
  if (trackingDisabledFromEnvVar) {
    enabled = false
  }
  
  trackingEnabled = enabled
```

We can simplify it to this

**Review by LekoArts — CHANGES_REQUESTED** [review]

**Review by LekoArts — APPROVED** [review]
🌮

**Comment by marvinjude** [comment]
Published in `create-gatsby@2.5.1`

**Comment by dhoko** [comment]
that was fast, thx :pray:
